src/resources/auth/auth.service.js

```diff
@@ -0,0 +1,84 @@
+const jwt = require('jsonwebtoken');
+const { validationResult } = require('express-validator');
+const HttpError = require('../../tools/httpError');
+const { register, login } = require('./auth.repository');
+const { users } = require('../../loader/dbconnect');
+const config = require('../../config/config');
+
+async function registerUser(req, res, next) {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      throw new HttpError('Registration error', 403, errors);
+    }
+    const {
+      email,
+      password,
+      role,
+      firstname,
+      lastname,
+      gender,
+      birthday,
+      residence,
+    } = req.body;
+
+    register({
+      email,
+      password,
+      role,
+      firstname,
+      lastname,
+      gender,
+      birthday,
+      residence,
+    });
+
+    res.json(users);
+  } catch (err) {
+    next(err);
+  }
+}
+
+async function loginUser(req, res, next) {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      throw new HttpError('Registration error', 403, errors);
+    }
+    console.log('REQ BODY', req.body);
```

Comment:
please remove console.log. Its good to use in development but not in production

Reply:
😄  I forgot remove it